website/source/docs/auth/aws-iam.html.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: "docs"
+page_title: "Auth Backend: AWS-IAM"
+sidebar_current: "docs-auth-aws-iam"
+description: |-
+  The aws-iam backend allows automated authentication AWS IAM principals.
+---
+
+# Auth Backend: aws-iam
+
+The aws-iam auth backend provides a secure introduction mechanism for AWS IAM
+principals. This allows you to reduce the problem of securely introducing a
+Vault token to the problem of securely introducing AWS IAM credentials, which
+AWS has already solved in a number of use cases, such as via EC2 Instance
+Profiles and IAM roles attached to Lambda functions. It also allows you to have
+a consistent workflow between developers working on a local laptop with tools
+such as Hologram.
+
+## Comparison with AWS-EC2 Authentication Backend
+
+The AWS-IAM and AWS-EC2 authentication backends serve similar purposes. Both
```

Comment:
AWS has responded to me confirming the above with, "That would be a breaking / backwards-incompatible change and there are no plans to do so. But the overall format including the instance id at the end is part of the current contract and is unlikely to ever change."

I also had a couple other thoughts along these lines for potential future enhancements.

One is that it would be great if `GetCallerIdentity` would return who the original session issuer was, so that you could see definitively if the credentials were handed out to an EC2 instance in an IAM instane profile (and no longer had to rely on `"Service": "ec2.amazonaws.com"` being the only `Principal` in the AssumeRolePolicyDocument as I mentioned above).

The second would be if `GetCallerIdentity` returned whether the credentials were MFA-authenticated (e.g., temporary credentials obtained by calling `GetSessionToken` with MFA authentication). This would allow you to only allow MFA-authenticated credentials to be able to authenticate to Vault.

Thoughts? Worth asking AWS?